### PR TITLE
move GetLocalArtBaseFilename to ArtUtils

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Move another function out of CFileItem and into ArtUtils

## Motivation and context
Less code in CFileItem

## How has this been tested?
It builds + I have added tests

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/video/VideoInfoScanner.cpp [hunk 1]
```diff
@@ -1792,14 +1793,17 @@ namespace KODI::VIDEO
...
         if (bApplyToDir && (content == CONTENT_MOVIES || content == CONTENT_MUSICVIDEOS))
         {
-          std::string filename = pItem->GetLocalArtBaseFilename();
+          std::string filename = ART::GetLocalArtBaseFilename(*pItem, useFolder);
           std::string directory = URIUtils::GetDirectory(filename);
           if (filename != directory)
             AddLocalItemArtwork(art, artTypes, directory, addAll, exactName);
         }
-        AddLocalItemArtwork(art, artTypes, pItem->GetLocalArtBaseFilename(), addAll, exactName);
+        useFolder = false;
```
Comment (garbear) [inline xbmc/video/VideoInfoScanner.cpp:1804]: Why is this line added? Maybe a comment is needed?

Comment (notspiff) [reply xbmc/video/VideoInfoScanner.cpp:1804]: it's to ensure the same behavior as earlier. notice that the call to GetLocalArtBaseFilename can mutate the bool parameter. since i'm reusing the same variable i need to reset it false in case it was mutated. earlier there was a convenience wrapper without the bool parameter which just threw it away. as it saw were limited use (basically, just here) i am of the opinion that bloating the api is a bigger problem than the slight convenience offered by the wrapper.

Comment (garbear) [reply xbmc/video/VideoInfoScanner.cpp:1804]: Agreed.

How about:

```patch
--- a/xbmc/video/VideoInfoScanner.cpp
+++ b/xbmc/video/VideoInfoScanner.cpp
@@ -1798,7 +1798,11 @@ namespace KODI::VIDEO
           if (filename != directory)
             AddLocalItemArtwork(art, artTypes, directory, addAll, exactName);
         }
+
+        // // Reset useFolder to false as GetLocalArtBaseFilename may modify it
+        // in the previous call.
         useFolder = false;
+
         AddLocalItemArtwork(art, artTypes, ART::GetLocalArtBaseFilename(*pItem, useFolder), addAll,
                             exactName);
       }
```

Comment (notspiff) [reply xbmc/video/VideoInfoScanner.cpp:1807]: sure.

**Review by ksooo — APPROVED** [review]
LGTM

**Review by garbear — APPROVED** [review]
Looks good to Ksooo
